DEVELOPER.md

```diff
@@ -1,58 +1,183 @@
+- Indy-SDK uses the ZMQ protocol to interact with the Indy ledgers. This might be blocked by some corporate firewalls as it's a non-standard protocol that doesn't use HTTP/s.
+- AFJ uses the HTTP protocol to communicate with Aries agents and WebSockets for messaging via a mediator.
+- Bifold relies on a mediator because mobile devices don't have a fixed IP address and often don't accept inbound network connections. The mediator, a service that runs on a server with a fixed IP address, relays messages between an agent and Bifold. The mediator is configured within the Bifold app.
+
+## Setup
+
+The setup for Bifold is similar to other React Native projects. The following sections will walk you through the process of setting up your development environment, installing dependencies, and running the app in an emulator.
+
+### Prerequisites
+
+This is a [yarn](https://yarnpkg.com) based project, not [npm](https://www.npmjs.com/). You will need to install yarn if you don't already have it installed. Also, you will need a version of node that is compatible with the version of yarn specified in the `engines` field of [./package.json](./package.json). If you don't have a compatible version of node installed, you can use [nvm](https://github.com/nvm-sh/nvm) to install a compatible version of node.
+
+```sh
+npm install -g yarn
+nvm install 16.15
+```
+
+This project will need to run on an iOS device or Android device or emulator. While it is recommended to test your software on both, especially if you're contributing back to the project, for demonstration purposes you can choose one or the other.
+
+[Android Studio](https://developer.android.com/studio)
+[Apple Xcode](https://developer.apple.com/xcode/)
+
+**ProTip 🤓**
 
 If you are using Mac OS with ARM64 chip, see this [suggested setup](./DEVELOPER_MACOS_arm64.md)
 
+### Setup & Configure for All Platforms
+
+Start by cloning the repository:
 
-## React Native setup
-React Native environment setup instructions are documented [here](https://reactnative.dev/docs/environment-setup). Be sure to select the correct React Native version (currently 0.66.x) from the dropdown. This will guide you through setting up your development environment for your operating system and choice of iOS (only if you are using a Mac) or Android. 
+```sh
+git clone https://github.com/hyperledger/aries-mobile-agent-react-native.git bifold \ &&
+cd bifold
+```
+
+Install all the package dependencies by running the following command from the root of the cloned repository:
 
-## Installing dependencies
-Next, install the npm modules needed to build and test the app.
 ```sh
-# from the root of the cloned repository, run:
 yarn install
 ```
 
-## Building packages
-Some packages need to be build, transpiled befoer it can be uses from the app
+Some packages need to be build, transpiled before it can be uses from the app. Do this with the following command:
```

Comment:
Some packages need to be *built*, transpiled before *they* can be *used* from the app.